src/utility/time.js

```diff
@@ -1,3 +1,20 @@
 import { zfill } from './string';
 
 export const getTimer = (number) => zfill(Math.floor(number / 60), 2) + ':' + zfill(number % 60, 2);
+
+/**
+ * Async delay that resolves after x ms.
+ *
+ * @example await sleep(TimeDuration.OneSecond);
+ * @param {number} ms Number of milliseconds to sleep.
+ * @returns
+ */
+export const sleep = (ms) => new Promise((resolve) => setTimeout(resolve, ms));
+
+/**
+ * Enum of common time durations in human readable format.
+ */
+export const TimeDuration = {
+	OneSecond: 1000,
+	TwoSeconds: 2000,
+};
```

Comment:
Good call re disabled button, I've pushed a grayscale version. This change also introduces a slight transition on the filter, let me know what you think:

![CleanShot 2021-12-03 at 15 09 07](https://user-images.githubusercontent.com/199204/144549145-66c30592-05a3-4e02-b174-46b535bb41ff.gif)

Reply:
Nice, replaced the constant with `Phaser.Timer.SECOND * 2`.

Reply:
It's very good progress, but I still meant that you can get rid of the stuff added to timer.js by using those Phaser functions as there's even an autoDestroy bolean and such. https://photonstorm.github.io/phaser-ce/Phaser.TimerEvent.html https://photonstorm.github.io/phaser-ce/Phaser.Timer.html - provided I'm not confused, I'm a bad coder after all :)

Reply:
I've removed the `TimeDuration` enum, but left the `sleep()` helper function. I haven't found an equivalent function in Phaser but let me know if I've missed something!

Reply:
Well, I'm trying to figure out how is this actually different overall compared to the delay from here: https://photonstorm.github.io/phaser-ce/Phaser.TimerEvent.html so that I have a better grasp of coding.

Reply:
I'm not sure `Phaser.TimerEvent` is suitable because it is intended for adding timed events to the main game loop and can be affected by slow motion, pausing, or other situations which affect the speed of the game loop.